agent/internal/fluent.go

```diff
@@ -445,3 +469,17 @@ func (f *fluentActor) Receive(ctx *actor.Context) error {
 	}
 	return nil
 }
+
+func (f *fluentActor) trackLogs(ctx *actor.Context) {
+	if err := trackLogs(ctx, f.docker, f.containerID, ctx.Self()); err != nil {
+		ctx.Log().Errorf("error tracking Fluent Bit logs: %s", err)
+	}
+	ctx.Log().Error("Fluent Bit failed, dumping all recent logs")
+	i0 := f.fluentLogsCount - len(f.fluentLogs)
+	if i0 < 0 {
+		i0 = 0
+	}
+	for i := i0; i < f.fluentLogsCount; i++ {
+		ctx.Log().Error(f.fluentLogs[i%len(f.fluentLogs)].Value)
```

Comment:
How do you feel about logging this all in one go to maybe make it more readable if agent logs get interleaved?

Reply:
These will all come out pretty much instantaneously and I'd rather try not to mess with the line-based output of the agent's own logs, so I'll keep it this way.

Reply:
Makes sense to me.